### DISCUSSION TITLE
HELP: clangd over tramp (compile_commands.json)

### DISCUSSION THREAD

**Discussion by Liorst4** [joaotavora/eglot#655] [Q&A] [answered]
When I try to use eglot with clangd on a remote project (tramp) I keep getting errors from clangd about it not being able to find compile_commands.json.

It tries too look in `/ssh:USER@HOST:/PATH`, and it can't find it (because its not running inside emacs).

I know that clangd supports getting the compile commands directory from the initialization options.
https://clangd.llvm.org/extensions.html#compilation-commands

What would be the best way to provide it a compile_commands.json ?

**Comment by joaotavora** [comment]
Hello @Liorst4 , I took the liberty of converting your issue to a GitHub "discussion".  You're the first one, but I will be doing a lot of this in the future, since I need the issue list to be more focused.  I hope someone can help you here (I can't at the moment, sorry).  @bjc implemented the TRAMP functionality, but it also seems that your problem is related to non-tramp things?

**Reply by Liorst4** [reply]
> Hello @Liorst4 , I took the liberty of converting your issue to a GitHub "discussion".

NP, wasn't aware of this feature in github.

> but it also seems that your problem is related to non-tramp things?

Not really, its only an issue in TRAMP.
In local files, `clangd` works fine.


I tried to copy the `compile_commands.json` and use it locally, but all the paths were invalid.
Adding `--path-mappings` might solve that (https://manpages.ubuntu.com/manpages/focal/man1/clangd.1.html).

Another possibility is maybe using `settings.compilationDatabaseChanges` (https://clangd.llvm.org/extensions.html#compilation-commands)

Maybe in the future the remote indexer feature could be used with TRAMP (https://clangd.llvm.org/remote-index.html)

**Comment by Liorst4** [comment, marked as answer]
My bad, apparently I was using version 1.7 from GNU elpa instead of the latest version from MELPA.
In the new version it just runs clangd on the remote machine, which solves the issue.
